File: frontend/src/types/inventory.ts
export interface InventoryBalance {
  lot_id: number;
  lot_code: string;
  product_id: number;
  product_name: string;
  unit: string;
  received_date: string;
  age_days: number;
  received_by: string;
  location_id: number;
  location_code: string;
  warehouse_code: string;
  warehouse_name: string;
  qty: number;
  total_qty: number;
}

export interface InventoryMovement {
  movement_id: number;
  lot_id: number;
  kind: string;
  from_location_code: string | null;
  to_location_code: string | null;
  qty: number;
  actor_name: string;
  note: string;
  adjustment_request_id: number | null;
  created_at: string;
}

File: backend/schemas/inventory.py
from pydantic import BaseModel


class InventoryBalance(BaseModel):
    lot_id: int
    lot_code: str
    product_id: int
    product_name: str
    unit: str
    received_date: str
    expires_on: str | None
    expiry_status: str
    age_days: int
    received_by: str
    location_id: int
    location_code: str
    warehouse_code: str
    warehouse_name: str
    qty: int
    total_qty: int


class InventoryMovement(BaseModel):
    movement_id: int
    lot_id: int
    kind: str
    from_location_code: str | None
    to_location_code: str | None
    qty: int
    actor_name: str
    note: str
    adjustment_request_id: int | None
    created_at: str
